### ISSUE TITLE
Each search query times out for Nominatim 2.5 running on a full Germany map

### ISSUE THREAD

**Issue by vak** [mediagis/nominatim-docker#31]
Synopsis: 
Each search query times out for Nominatim 2.5 running on a full Germany map.  

Details: 
I am trying to use nominatim-docker approach to create a working Nominatim docker image for Germany. After several building days the image was created (about 90Gb), but I am having difficulties to perform any query. Here are all the Postgres log entries after starting the image in docker container and making a query from Nominatim web interface:

```
2017-12-11 18:23:43 UTC LOG:  database system was shut down at 2017-12-04 07:51:31 UTC
2017-12-11 18:23:43 UTC LOG:  MultiXact member wraparound protections are now enabled
2017-12-11 18:23:43 UTC LOG:  database system is ready to accept connections
2017-12-11 18:23:43 UTC LOG:  autovacuum launcher started
2017-12-11 18:23:44 UTC LOG:  incomplete startup packet
2017-12-11 18:25:35 UTC ERROR:  canceling statement due to statement timeout
2017-12-11 18:25:35 UTC STATEMENT:  select place_id, 0::int as exactmatch from search_name where name_vector @> ARRAY[2513074] order by (case when importance = 0 OR importance IS NULL then 0.75-(search_rank::float/40) else importance end) DESC limit 20
```

CPU usage is almost zero, RAM usage is also tiny. Nothing seems much to happen before and after the request is done. During the request is done i see at least one of 8 cores sometimes loaded up to 50% by a postgres process

The server has 122 Gb RAM and like 80Gb disk space free for use (checked from being within a docker container)

btw, is there some way to "warm-up" RAM and preload it with Postgres' OSM map content ?

Hints are welcome!

**Comment by mikeppp** [comment]
Have the exact same problem running all of North America.

**Comment by jarbaugh5** [comment]
@vak @mikeppp were you able to solve this? running into the same problem with North America as well

**Comment by mikeppp** [comment]
No I haven't!!!
We are kind of dead in the water until this resolved :(

EVERY flavour of nominatum docker we have tried has the SAME issue!! 

I can't believe that this has been such an issue for so long with no one fixing it!!

**Comment by jarbaugh5** [comment]
Darn.. Thanks for the reply! Looks like I may have to use a paid API to reliably geocode without a ton of hassle

**Comment by mikeppp** [comment]
Yup :(

If you use Azure you can Bing for relatively cheaply.... It's quite fast and North America geocoding coverage is excellent...

**Comment by jarbaugh5** [comment]
Oh cool - I do use azure, I'll take a look!

**Comment by vak** [comment]
yes. I resolved it somehow. if I remember right, i've entered the Postgres and just re-issued the last query shown in logs from console. After this query succeeded things were fine with Nominatim. However I don't remember the details, it was in Dec 2017 ;)

**Comment by mikeppp** [comment]
Yes it worked for us as well in 2017 ...

Try it now and see how it goes...

**Comment by Gitii** [comment]
Maybe it is because of the Postgres config?
I experienced similar errors:
I entered a street name, the server started thinking and after a several seconds, an error page was displayed (server error, internal timeout).
After that I took a look into the Postgres config (which is all default). I copied this

> shared_buffers = 1GB # min 128kB    
maintenance_work_mem = 10GB # min 1MB    
work_mem = 50MB # min 64kB    
maintenance_work_mem = 1GB # min 1MB    
effective_cache_size = 1GB    
synchronous_commit = off # synchronization level;  
checkpoint_segments = 10 # in logfile segments, min 1, 16MB each  
checkpoint_timeout = 10min # range 30s-1h  
checkpoint_completion_target = 0.9 # checkpoint target duration, 0.0 - 1.0  

at the end of config file and restarted everything.
I have not experienced any issues after that (quick test using some random adresses in germany).

Hopefully this "fixed" it. But it just might working because of hot caches.
